packages/calcite-components/src/components/dropdown/dropdown.tsx

```diff
@@ -634,30 +637,23 @@ export class Dropdown
     }
   }
 
-  private focusOnFirstActiveOrFirstItem = (): void => {
-    this.getFocusableElement(
-      this.getTraversableItems().find((item) => item.selected) || this.items[0],
-    );
-  };
+  private focusOnFirstActiveOrDefaultItem = (): void => {
+    const selectedItem = this.getTraversableItems().find((item) => item.selected);
+    const target: HTMLCalciteDropdownItemElement =
+      selectedItem ||
+      (this.focusLastDropdownItem ? this.items[this.items.length - 1] : this.items[0]);
 
-  private getFocusableElement(item: HTMLCalciteDropdownItemElement): void {
-    if (!item) {
+    this.focusLastDropdownItem = false;
+
+    if (!target) {
       return;
     }
 
-    focusElement(item);
-  }
-
-  private toggleOpenEnd = (): void => {
-    this.focusOnFirstActiveOrFirstItem();
-    this.el.removeEventListener("calciteDropdownOpen", this.toggleOpenEnd);
+    focusElement(target);
   };
 
-  private openCalciteDropdown = () => {
+  private clickHandler = (): void => {
```

Comment:
Can we keep this dedicated click handler and restore the previous `openCalciteDropdown` (name could be simplified to `openDropdown`)? That way, other event handlers can call this dedicated open method instead of invoking an event handler for a different event.

Reply:
i think `clickHandler( )` isn't required and we can use `toggleDropdown` (instead of openDropdown)  method to handle click.